frontend/src/components/global/AlgoDraggable.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+    <div :id="id" class="algo-draggable-container">
+        <TransitionGroup name="algo-draggable-group">
+            <v-chip
+                closable
+                v-for="variable in this.selectedVariables"
+                class="ma-2"
+                v-bind:class="{ dragged: draggedVariableId === variable.id }"
+                :key="variable.id"
+                @click:close="deleteVariable(variable)"
+                draggable
+                style="cursor: grab"
+                v-on:dragstart="dragStart"
+                v-on:dragenter="dragEnter"
+                v-on:dragend="dragEnd"
+                :id="variable.id"
+            >
+                {{ variable.name }}
+            </v-chip>
+        </TransitionGroup>
+    </div>
+</template>
+
+<script>
+    import { defineComponent } from "vue";
+
+    export default defineComponent({
+        props: ["id", "draggableList"],
+
+        data() {
+            return {
+                selectedVariables: [],
+                draggedVariableId: "",
+            };
+        },
+
+        mounted() {
+            this.selectedVariables = this.$props.draggableList;
+        },
+
+        methods: {
+            deleteVariable(selectedVariable) {
+                let index = this.selectedVariables.findIndex((variable) => variable.id === selectedVariable.id);
+                if (index != -1) this.selectedVariables.splice(index, 1);
+            },
+
+            dragStart: function (event) {
+                this.draggedVariableId = event.target.id;
+            },
+
+            dragEnter: function (event) {
+                if (
+                    event.target.id != this.draggedVariableId &&
+                    event.target.parentNode.classList.contains("algo-draggable-container") &&
+                    !event.target.classList.contains("algo-draggable-group-move")
+                ) {
+                    let index_from = this.selectedVariables.findIndex(
+                        (variable) => variable.id === this.draggedVariableId
+                    );
+                    let index_to = this.selectedVariables.findIndex((variable) => variable.id === event.target.id);
+                    let cutOut = this.selectedVariables.splice(index_from, 1)[0];
+                    this.selectedVariables.splice(index_to, 0, cutOut);
+                }
+            },
+
+            dragEnd: function () {
+                this.draggedVariableId = "";
+            },
+        },
+    });
+</script>
+
+<style scoped>
+    .algo-draggable-container {
+        height: 15%;
+        white-space: nowrap;
+        overflow-y: auto;
+    }
+
+    .dragged {
+        transform: translate(0px, -5px);
```

Comment:
Znalazłem winowajcę to ta linijka `transform: translate(0px, -5px);` sprawia, że się pojawia ten biały prostokąt. Jak to usunąłem to wyświetla się normalnie ten przesuwany chip. Ja bym z tego po prostu zrezygnował. Usuń tę klasę i jej bindowanie w htmlu.

P.S. Nie będę rzucał nazwiskami, ale to pewien student Sukasz Łkabowski zaproponował Ci to podnoszenie chipa, które spowodowało pojawienie się tego białego prostokąta xD. Ja bym na niego zwalił winę xDD